lib/new_relic/dependency_detection.rb

```diff
@@ -179,6 +179,15 @@ def fetch_config_value(key)
       return valid_value
     end
 
+    # update any :auto config value to be either :prepend or :chain after auto
+    # determination has selected one of those to use
+    def update_config_value(use_prepend)
+      if config_key && auto_configured?
+        NewRelic::Agent.config.instance_variable_get(:@cache)[config_key] = use_prepend ? :prepend : :chain
+      end
+      use_prepend
```

Comment:
Is this saying we are always returning `true` or `false`?

Reply:
Precisely. We hook into this existing method that returns a boolean, so here we make sure we still always return a boolean.

Reply:
Got it I see what is happening now. Thank you :)